ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx

```diff
@@ -253,26 +279,70 @@ export function PublishOnDemandWidget(props: PublishOnDemandWidgetProps) {
     });
   };
 
+  const onSubmitPublishEverything = () => {
+    console.log('publish everything!', publishEverythingFormData);
+    setIsSubmitting(true);
+    const { publishingTarget, comment } = publishEverythingFormData;
+    publishAll(siteId, publishingTarget, comment).subscribe({
+      next() {
+        setIsSubmitting(false);
+        dispatch(
+          showSystemNotification({
+            message: formatMessage(messages.publishSuccess)
+          })
+        );
+        setPublishEverythingFormData({ ...initialPublishEverythingFormData, publishingTarget });
+        setMode(null);
+      },
+      error({ response }) {
+        setIsSubmitting(false);
+        dispatch(
+          showSystemNotification({
+            message: response.message,
+            options: { variant: 'error' }
+          })
+        );
+      }
+    });
+  };
+
   const onCancel = () => {
     setMode(null);
     setDefaultPublishingTarget(publishingTargets, true);
   };
 
   useEffect(() => {
-    if (mode === 'studio') {
-      setPublishStudioFormValid(
-        publishStudioFormData.path.replace(/\s/g, '') !== '' &&
-          publishStudioFormData.publishingTarget !== '' &&
-          (!bulkPublishCommentRequired || !isBlank(publishStudioFormData.comment))
-      );
-    } else {
-      setPublishGitFormValid(
-        publishGitFormData.commitIds.replace(/\s/g, '') !== '' &&
-          publishGitFormData.publishingTarget !== '' &&
-          (!publishByCommitCommentRequired || !isBlank(publishGitFormData.comment))
-      );
+    switch (mode) {
+      case 'studio':
+        setPublishStudioFormValid(
+          publishStudioFormData.path.replace(/\s/g, '') !== '' &&
+            publishStudioFormData.publishingTarget !== '' &&
+            (!bulkPublishCommentRequired || !isBlank(publishStudioFormData.comment))
+        );
+        break;
+      case 'git':
+        setPublishGitFormValid(
+          publishGitFormData.commitIds.replace(/\s/g, '') !== '' &&
+            publishGitFormData.publishingTarget !== '' &&
+            (!publishByCommitCommentRequired || !isBlank(publishGitFormData.comment))
+        );
+        break;
+      case 'all':
+        setPublishEverythingFormValid(
+          publishEverythingFormData.publishingTarget !== '' &&
+            (!publishEverythingCommentRequired || !isBlank(publishEverythingFormData.comment))
+        );
+        break;
     }
-  }, [publishStudioFormData, publishGitFormData, mode, bulkPublishCommentRequired, publishByCommitCommentRequired]);
+  }, [
+    publishStudioFormData,
+    publishGitFormData,
+    publishEverythingFormData,
+    mode,
+    bulkPublishCommentRequired,
+    publishByCommitCommentRequired,
+    publishEverythingCommentRequired
+  ]);
```

Comment:
This looks like it's executed too often and it's not too expensive. Let's just ditch the effect and the `useState` and calculate the `isValid` on the fly.